### ISSUE TITLE
Unable to build

### ISSUE THREAD

**Issue by vadi2** [edbee/edbee-app#1]
Hi,

I can't get the app to build:

```
20:18:31: Running steps for project edbee-app...
20:18:31: Starting: "/usr/bin/make" clean
cd edbee-lib/ && ( test -e Makefile || /home/vadi/Programs/Qt/5.8/gcc_64/bin/qmake -o Makefile /home/vadi/Programs/edbee-app/edbee-lib/edbee-lib.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile clean
make[1]: Entering directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib'
cd edbee-lib/ && ( test -e Makefile || /home/vadi/Programs/Qt/5.8/gcc_64/bin/qmake -o Makefile /home/vadi/Programs/edbee-app/edbee-lib/edbee-lib/edbee-lib.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile clean
make[2]: Entering directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib/edbee-lib'
rm -f moc_predefs.h
rm -f moc_texteditorwidget.cpp moc_textrenderer.cpp moc_textdocument.cpp moc_chartextdocument.cpp moc_texteditorconfig.cpp moc_textundostack.cpp moc_texteditorcontroller.cpp moc_textrange.cpp moc_textdocumentscopes.cpp moc_textlinedata.cpp moc_textbuffer.cpp moc_test.cpp moc_texteditorcommandmap.cpp moc_texteditorcomponent.cpp moc_textmargincomponent.cpp moc_texttheme.cpp moc_texteditorscrollarea.cpp moc_textsearcher.cpp moc_edbee.cpp
rm -f debug_new.o debug_allocs.o simpleprofiler.o textcodecdetector.o lineending.o texteditorwidget.o textrenderer.o textdocument.o chartextdocument.o texteditorconfig.o texteditorkeymap.o textundostack.o textcaretcache.o textlexer.o textrange.o textselection.o textdocumentscopes.o grammartextlexer.o lineoffsetvector.o textlinedata.o textbuffer.o chartextbuffer.o texteditorcontroller.o texteditorcommand.o selectioncommand.o undocommand.o redocommand.o replaceselectioncommand.o copycommand.o cutcommand.o pastecommand.o textdocumentserializer.o test.o textcodec.o tmlanguageparser.o debugcommand.o regexp.o tmthemeparser.o baseplistparser.o jsonparser.o textgrammar.o texteditorcommandmap.o texteditorcomponent.o texteditorrenderer.o textmargincomponent.o texttheme.o texteditorscrollarea.o textsearcher.o findcommand.o keymapparser.o tabcommand.o edbee.o textdocumentfilter.o cascadingqvariantmap.o duplicatecommand.o newlinecommand.o util.o removecommand.o change.o abstractrangedchange.o linedatalistchange.o linedatachange.o selectionchange.o textchange.o textchangewithcaret.o mergablechangegroup.o commentcommand.o rangesetlineiterator.o dynamicvariables.o rangelineiterator.o factorykeymap.o factorycommandmap.o moc_texteditorwidget.o moc_textrenderer.o moc_textdocument.o moc_chartextdocument.o moc_texteditorconfig.o moc_textundostack.o moc_texteditorcontroller.o moc_textrange.o moc_textdocumentscopes.o moc_textlinedata.o moc_textbuffer.o moc_test.o moc_texteditorcommandmap.o moc_texteditorcomponent.o moc_textmargincomponent.o moc_texttheme.o moc_texteditorscrollarea.o moc_textsearcher.o moc_edbee.o
rm -f *~ core *.core
make[2]: Leaving directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib/edbee-lib'
cd edbee-test/ && ( test -e Makefile || /home/vadi/Programs/Qt/5.8/gcc_64/bin/qmake -o Makefile /home/vadi/Programs/edbee-app/edbee-lib/edbee-test/edbee-test.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile clean
make[2]: Entering directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib/edbee-test'
rm -f moc_predefs.h
rm -f moc_replaceselectioncommandtest.cpp moc_textrangetest.cpp moc_textdocumenttest.cpp moc_textbuffertest.cpp moc_textlinedatatest.cpp moc_gapvectortest.cpp moc_lineoffsetvectortest.cpp moc_lineendingtest.cpp moc_textdocumentserializertest.cpp moc_tmlanguageparsertest.cpp moc_regexptest.cpp moc_textdocumentscopestest.cpp moc_textundostacktest.cpp moc_cascadingqvariantmaptest.cpp moc_textsearchertest.cpp moc_duplicatecommandtest.cpp moc_newlinecommandtest.cpp moc_utiltest.cpp moc_grammartextlexertest.cpp moc_removecommandtest.cpp moc_linedatalistchangetest.cpp moc_textchangetest.cpp moc_mergablechangegrouptest.cpp moc_rangesetlineiteratortest.cpp moc_dynamicvariablestest.cpp moc_rangelineiteratortest.cpp
rm -f replaceselectioncommandtest.o textrangetest.o textdocumenttest.o textbuffertest.o textlinedatatest.o gapvectortest.o lineoffsetvectortest.o main.o lineendingtest.o textdocumentserializertest.o tmlanguageparsertest.o regexptest.o textdocumentscopestest.o textundostacktest.o cascadingqvariantmaptest.o textsearchertest.o duplicatecommandtest.o newlinecommandtest.o utiltest.o grammartextlexertest.o removecommandtest.o linedatalistchangetest.o textchangetest.o mergablechangegrouptest.o rangesetlineiteratortest.o dynamicvariablestest.o rangelineiteratortest.o moc_replaceselectioncommandtest.o moc_textrangetest.o moc_textdocumenttest.o moc_textbuffertest.o moc_textlinedatatest.o moc_gapvectortest.o moc_lineoffsetvectortest.o moc_lineendingtest.o moc_textdocumentserializertest.o moc_tmlanguageparsertest.o moc_regexptest.o moc_textdocumentscopestest.o moc_textundostacktest.o moc_cascadingqvariantmaptest.o moc_textsearchertest.o moc_duplicatecommandtest.o moc_newlinecommandtest.o moc_utiltest.o moc_grammartextlexertest.o moc_removecommandtest.o moc_linedatalistchangetest.o moc_textchangetest.o moc_mergablechangegrouptest.o moc_rangesetlineiteratortest.o moc_dynamicvariablestest.o moc_rangelineiteratortest.o
rm -f *~ core *.core
make[2]: Leaving directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib/edbee-test'
make[1]: Leaving directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib'
cd edbee-app/ && ( test -e Makefile || /home/vadi/Programs/Qt/5.8/gcc_64/bin/qmake -o Makefile /home/vadi/Programs/edbee-app/edbee-app/edbee-app.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile clean
make[1]: Entering directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-app'
rm -f qrc_main.cpp qrc_QtAwesome.cpp
rm -f moc_predefs.h
rm -f moc_mainwindow.cpp moc_filetreesidewidget.cpp moc_application.cpp moc_gotowidget.cpp moc_findwidget.cpp moc_windowmanager.cpp moc_workspace.cpp moc_tabwidget.cpp moc_workspacemanager.cpp moc_QtAwesome.cpp
rm -f main.o mainwindow.o filetreesidewidget.o application.o gotowidget.o findwidget.o edbeeconfig.o windowmanager.o workspace.o workspaceserializer.o revealinsidebarcommand.o fileutil.o tabwidget.o workspacemanager.o appstateserializer.o QsLogDest.o QsLog.o QsDebugOutput.o QtAwesome.o qrc_main.o qrc_QtAwesome.o moc_mainwindow.o moc_filetreesidewidget.o moc_application.o moc_gotowidget.o moc_findwidget.o moc_windowmanager.o moc_workspace.o moc_tabwidget.o moc_workspacemanager.o moc_QtAwesome.o
rm -f -r edbee.app
rm -f *~ core *.core
make[1]: Leaving directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-app'
cd edbee-app-test/ && ( test -e Makefile || /home/vadi/Programs/Qt/5.8/gcc_64/bin/qmake -o Makefile /home/vadi/Programs/edbee-app/edbee-app-test/edbee-app-test.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile clean
make[1]: Entering directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-app-test'
rm -f moc_predefs.h
rm -f moc_edbeeconfigtest.cpp
rm -f main.o edbeeconfigtest.o edbeeconfig.o QsLogDest.o QsLog.o QsDebugOutput.o moc_edbeeconfigtest.o
rm -f *~ core *.core
make[1]: Leaving directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-app-test'
20:18:32: The process "/usr/bin/make" exited normally.
20:18:32: Configuration unchanged, skipping qmake step.
20:18:32: Starting: "/usr/bin/make" 
cd edbee-lib/ && ( test -e Makefile || /home/vadi/Programs/Qt/5.8/gcc_64/bin/qmake -o Makefile /home/vadi/Programs/edbee-app/edbee-lib/edbee-lib.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile 
make[1]: Entering directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib'
cd edbee-lib/ && ( test -e Makefile || /home/vadi/Programs/Qt/5.8/gcc_64/bin/qmake -o Makefile /home/vadi/Programs/edbee-app/edbee-lib/edbee-lib/edbee-lib.pro -spec linux-g++ CONFIG+=debug CONFIG+=qml_debug ) && /usr/bin/make -f Makefile 
make[2]: Entering directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib/edbee-lib'
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_NODLL -DQT_QML_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I../../../edbee-app/edbee-lib/edbee-lib -I. -I../../../edbee-app/edbee-lib/edbee-lib -I../../../edbee-app/edbee-lib/edbee-lib -I../../../Qt/5.8/gcc_64/include -I../../../Qt/5.8/gcc_64/include/QtWidgets -I../../../Qt/5.8/gcc_64/include/QtGui -I../../../Qt/5.8/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I../../../Qt/5.8/gcc_64/mkspecs/linux-g++ -o debug_new.o ../../../edbee-app/edbee-lib/edbee-lib/edbee/util/mem/debug_new.cpp
../../../edbee-app/edbee-lib/edbee-lib/edbee/util/mem/debug_new.cpp: In function ‘void operator delete [](void*, const char*, int)’:
../../../edbee-app/edbee-lib/edbee-lib/edbee/util/mem/debug_new.cpp:105:14: warning: deleting ‘void*’ is undefined [-Wdelete-incomplete]
     delete[] p;
              ^
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_NODLL -DQT_QML_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I../../../edbee-app/edbee-lib/edbee-lib -I. -I../../../edbee-app/edbee-lib/edbee-lib -I../../../edbee-app/edbee-lib/edbee-lib -I../../../Qt/5.8/gcc_64/include -I../../../Qt/5.8/gcc_64/include/QtWidgets -I../../../Qt/5.8/gcc_64/include/QtGui -I../../../Qt/5.8/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I../../../Qt/5.8/gcc_64/mkspecs/linux-g++ -o debug_allocs.o ../../../edbee-app/edbee-lib/edbee-lib/edbee/util/mem/debug_allocs.cpp
g++ -c -pipe -g -fPIC -std=gnu++11 -Wall -W -D_REENTRANT -DQT_NODLL -DQT_QML_DEBUG -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_CORE_LIB -I../../../edbee-app/edbee-lib/edbee-lib -I. -I../../../edbee-app/edbee-lib/edbee-lib -I../../../edbee-app/edbee-lib/edbee-lib -I../../../Qt/5.8/gcc_64/include -I../../../Qt/5.8/gcc_64/include/QtWidgets -I../../../Qt/5.8/gcc_64/include/QtGui -I../../../Qt/5.8/gcc_64/include/QtCore -I. -isystem /usr/include/libdrm -I../../../Qt/5.8/gcc_64/mkspecs/linux-g++ -o simpleprofiler.o ../../../edbee-app/edbee-lib/edbee-lib/edbee/util/simpleprofiler.cpp
In file included from ../../../edbee-app/edbee-lib/edbee-lib/edbee/util/simpleprofiler.cpp:6:0:
../../../edbee-app/edbee-lib/edbee-lib/edbee/util/simpleprofiler.h:8:20: fatal error: config.h: No such file or directory
compilation terminated.
Makefile:2572: recipe for target 'simpleprofiler.o' failed
make[2]: Leaving directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib/edbee-lib'
Makefile:43: recipe for target 'sub-edbee-lib-make_first' failed
make[1]: Leaving directory '/home/vadi/Programs/build-edbee-app-Desktop_Qt_5_8_0_GCC_64bit-Debug/edbee-lib'
Makefile:44: recipe for target 'sub-edbee-lib-make_first' failed
make[2]: *** [simpleprofiler.o] Error 1
make[1]: *** [sub-edbee-lib-make_first] Error 2
make: *** [sub-edbee-lib-make_first] Error 2
20:18:33: The process "/usr/bin/make" exited with code 2.
Error while building/deploying project edbee-app (kit: Desktop Qt 5.8.0 GCC 64bit)
When executing step "Make"
20:18:33: Elapsed time: 00:02.
```

There doesn't seem to be a config.h in https://github.com/edbee/edbee-lib/tree/bef5c5fca265e8c7527fda353db173fc4c686391/edbee-lib/edbee/util as it is looking.

How can I fix this?

**Comment by gamecreature** [comment]
You're completely right about that. I guess this creeped in because we use a config.h in our project for configuring specific options. And this file refers to it.
It is used to enable and disable profile via the config.h file...
I will commit a new version without this config.h include.

**Comment by vadi2** [comment]
Yeah, that'd be good.

We're looking at switching [Mudlet](http://mudlet.org) over to edbee https://github.com/Mudlet/Mudlet/pull/983/files so I was hoping to try the app out to see what would it be like.

**Comment by gamecreature** [comment]
Two days ago I've already committed a version without this include:
https://github.com/edbee/edbee-lib/commit/5449d34a27bf946ac1a168a84c4907be884f2c86

**Comment by vadi2** [comment]
Thanks for the quick turnaround!

The other issue on a fresh clone is:

```
Could not find qmake spec 'default'.
Error while parsing file /home/vadi/Programs/edbee-app/edbee-app.pro. Giving up.
Cannot read /home/vadi/Programs/edbee-app/edbee-lib/vendor/QsLog/QsLog.pri: No such file or directory
```

I'm on Ubuntu, so the filesystem is case-sensitive - actual directory is `qslog` and not `QsLog`.

**Comment by vadi2** [comment]
Another similar error is `fileutil.cpp` including `QPRocess` instead of `QProcess`. With those two case-sensitive issues fixed, the build completes on Qt 5.6. Doesn't work on 5.8 because of some additional functions that were introduced and the compiler doesn't know which one to pick.

Thanks for the help!
